runbotics/runbotics-orchestrator-ui/src/src-app/components/BotProcessRunner.tsx

```diff
@@ -158,81 +169,159 @@ const BotProcessRunner: FC<BotProcessRunnerProps> = ({
                         variant: 'error',
                     }
                 );
+            })
+            .finally(() => {
+                dispatch(processInstanceActions.removeFromJobsMap({ processId }));
             });
     };
 
+    const handleRemoveJob = () => {
+        const job = jobsMap[processId];
+        switch (job.eventType) {
```

Comment:
It will look a bit cleaner if you change it to if statement. Please use switch case if we have many states to handle
const job = jobsMap[processId];

if (!job) return;
const eventType = job.eventType;
if (eventType === WsMessage.Activ || waiting) {
...
}